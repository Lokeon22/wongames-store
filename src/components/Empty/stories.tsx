import { StoryFn, Meta } from "@storybook/react"
import Empty, { EmptyProps } from "."

export default {
  title: "Empty",
  component: Empty,
  args: {
    title: "Nothing to see here",
    description: "Your wishlist is empty",
    hasLink: true
  },
  parameters: {
    backgrounds: {
      default: "won-dark"
    }
  }
} as Meta

export const Default: StoryFn<EmptyProps> = (args) => (
  <div style={{ maxWidth: "30rem", margin: "0 auto" }}>
    <Empty {...args} />
  </div>
)
