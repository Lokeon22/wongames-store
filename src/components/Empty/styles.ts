import styled, { css } from "styled-components"

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
`

export const Image = styled.img`
  max-width: 100%;
`

export const Title = styled.h2`
  ${({ theme }) => css`
    color: ${theme.colors.primary};
    font-size: ${theme.font.sizes.xxlarge};
    font-weight: ${theme.font.bold};
  `}
`

export const Description = styled.span`
  ${({ theme }) => css`
    color: ${theme.colors.gray};
    font-size: ${theme.font.sizes.xlarge};
    font-weight: ${theme.font.light};
    margin-bottom: ${theme.spacings.medium};
  `}
`
