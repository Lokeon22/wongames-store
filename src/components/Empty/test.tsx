import { screen } from "@testing-library/react"
import { renderWithTheme } from "../../utils/tests/helpers"

import Empty from "."

const props = {
  title: "Title Empty",
  description: "Description empty"
}

describe("<Empty />", () => {
  it("should render the Empty component", () => {
    const { container } = renderWithTheme(<Empty {...props} hasLink />)

    expect(
      screen.getByRole("img", { name: /a gamer in a couch playing/i })
    ).toBeInTheDocument()

    expect(
      screen.getByRole("heading", { name: /title empty/i })
    ).toBeInTheDocument()
    expect(screen.getByText("Description empty")).toBeInTheDocument()

    expect(
      screen.getByRole("button", { name: "Go back to store" })
    ).toBeInTheDocument()

    expect(container.parentElement).toMatchSnapshot()
  })

  it("should not render the link when is not passed", () => {
    renderWithTheme(<Empty {...props} />)

    expect(
      screen.queryByRole("button", { name: "Go back to store" })
    ).not.toBeInTheDocument()
  })
})
