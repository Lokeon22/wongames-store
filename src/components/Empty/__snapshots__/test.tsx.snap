// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Empty /> should render the Empty component 1`] = `
.c0 {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.c1 {
  max-width: 100%;
}

.c2 {
  color: #F231A5;
  font-size: 2.8rem;
  font-weight: 600;
}

.c3 {
  color: #8F8F8F;
  font-size: 2.0rem;
  font-weight: 300;
  margin-bottom: 3.2rem;
}

.c4 {
  display: inline-flex;
  align-items: center;
  justify-content: center;
  background: linear-gradient(189deg, #ff5f5f 0%, #f062c0 50%);
  border-radius: 0.4rem;
  border: 0;
  cursor: pointer;
  color: #FAFAFA;
  padding: 0.8rem;
  text-decoration: none;
  height: 4rem;
  font-size: 1.4rem;
  padding: 0.8rem 3.2rem;
}

.c4:hover {
  background: linear-gradient(180deg, #e35565 0%, #d958a6 50%);
}

<body>
  <div>
    <div
      class="c0"
    >
      <img
        alt="a gamer in a couch playing"
        class="c1"
        src="/img/empty.svg"
      />
      <h2
        class="c2"
      >
        Title Empty
      </h2>
      <span
        class="c3"
      >
        Description empty
      </span>
      <a
        href="/"
      >
        <button
          class="c4"
        >
          <span>
            Go back to store
          </span>
        </button>
      </a>
    </div>
  </div>
</body>
`;
